code/modules/library/book.dm

```diff
@@ -255,13 +255,15 @@
 			if(!length(pages))
 				to_chat(user, "<span class='notice'>There aren't any pages in this book!</span>")
 				return
-			var/page_choice = stripped_input(user, "There are [length(pages)] pages, which page number would you like to remove?")
+			var/page_choice = text2num(stripped_input(user, "There are [length(pages)] pages, which page number would you like to remove?"))
 			if(!page_choice)
 				to_chat(user, "<span class='notice'>You change your mind.</span>")
 				return
 			if(!isnum(page_choice) || page_choice <= 0 || page_choice > length(pages))
 				to_chat(user, "<span class='notice'>That is not an acceptable value.</span")
 				return
+			if(page_choice == length(pages))
+				current_page--
 			pages -= pages[page_choice]
```

Comment:
```suggestion
			remove_page(page_choice)
```
I think I forgot to fully implement page removals. But you can just call this proc on the book to remove the page. That proc itself has another error in it which could be fixed here aswell:
```dm
page_number = length(pages)
```
should be
```dm
current_page = min(current_page, length(pages))
```

Reply:
I really should've read the file more carefully, eh?
I also removed a redundant check from `remove_page()` while at it. Hopefully it won't cause any problems down the line.